src/test/UniswapV3SplitRouteCallee.t.sol

```diff
@@ -0,0 +1,375 @@
+        uint256 amount = 10 * WAD;
+        assertEq(link.balanceOf(address(this)), 0);
+        getLink(amount);
+        assertEq(link.balanceOf(address(this)), amount);
+    }
+
+    function joinLink(uint256 amount) private {
+        link.approve(linkJoinAddr, amount);
+        linkJoin.join(aliAddr, amount);
+    }
+
+    function testJoinLink() public {
+        uint256 amountLink = 100 * WAD;
+        getLink(amountLink);
+        joinLink(amountLink);
+        assertEq(vat.gem(linkName, aliAddr), amountLink);
+    }
+
+    function frobMax(uint256 gem, bytes32 ilkName) private {
+        uint256 ink = gem;
+        (, uint256 rate, uint256 spot,,) = vat.ilks(ilkName);
+        uint256 art = (ink * spot) / rate;
+        vat.frob(ilkName, aliAddr, aliAddr, aliAddr, int256(ink), int256(art));
+    }
+
+    function testFrobMax() public {
+        (,,,, uint256 dustRad) = vat.ilks(linkName);
+        uint256 amountLink = (dustRad / getLinkPriceRay()) * 2;
+        getLink(amountLink);
+        joinLink(amountLink);
+        frobMax(amountLink, linkName);
+        try vat.frob(linkName, aliAddr, aliAddr, aliAddr, 0, 1) {
+            log("not at maximum frob");
+            fail();
+        } catch {
+            log("success");
+        }
+    }
+
+    function drip(bytes32 ilkName) private {
+        jug.drip(ilkName);
+    }
+
+    function testDrip() public {
+        (, uint256 ratePre,,,) = vat.ilks(linkName);
+        drip(linkName);
+        (, uint256 ratePost,,,) = vat.ilks(linkName);
+        assertGt(ratePost, ratePre);
+    }
+
+    function barkLink() private returns (uint256 auctionId) {
+        dog.bark(linkName, aliAddr, aliAddr);
+        auctionId = linkClip.kicks();
+    }
+
+    function trimFunctionHash(bytes calldata data) public pure returns (bytes calldata, bytes calldata) {
+        return (data[:4], data[4:]);
+    }
+
+    function buildCalldata(bytes calldata signature, bytes calldata data) public pure returns (bytes memory) {
+        return abi.encodePacked(signature, data);
+    }
+
+    function reencodedAutoRouterData(bytes memory txData) private view returns (bytes memory output) {
+        (uint256 deadline, bytes[] memory calls) = abi.decode(txData, (uint256, bytes[]));
+        bytes[] memory transformedCalls = new bytes[](calls.length);
+        for (uint256 i = 0; i < calls.length; i++) {
+            (bytes memory signature, bytes memory callWithTrimmedHash) = this.trimFunctionHash(calls[i]);
+            UniswapV3ExactInputParams memory mCalldata = abi.decode(callWithTrimmedHash, (UniswapV3ExactInputParams));
+            UniswapV3ExactInputParams memory modifiedParams = UniswapV3ExactInputParams({
+                path: mCalldata.path,
+                recipient: address(dan),
+                amountIn: mCalldata.amountIn,
+                amountOutMinimum: mCalldata.amountOutMinimum
+            });
+            transformedCalls[i] = this.buildCalldata(signature, abi.encode(modifiedParams));
```

Comment:
Is there a reason for making it an external call?

Reply:
Adjusted.